test/framework/machinedeployment_helpers.go

```diff
@@ -448,6 +449,41 @@ func machineDeploymentOptions(deployment clusterv1.MachineDeployment) []client.L
 	}
 }
 
+// DeleteAndWaitMachineDeploymentInput is the input for DeleteAndWaitMachineDeployment.
+type DeleteAndWaitMachineDeploymentInput struct {
+	ClusterProxy              ClusterProxy
+	Cluster                   *clusterv1.Cluster
+	MachineDeployment         *clusterv1.MachineDeployment
+	WaitForMachineDeployments []interface{}
+	DeletePropagationPolicy   *metav1.DeletionPropagation
+}
+
+// DeleteAndWaitMachineDeployment deletes MachineDeployment and waits until it is gone.
+func DeleteAndWaitMachineDeployment(ctx context.Context, input DeleteAndWaitMachineDeploymentInput) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for DeleteAndWaitMachineDeployment")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling DeleteAndWaitMachineDeployment")
+	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling DeleteAndWaitMachineDeployment")
+	Expect(input.MachineDeployment).ToNot(BeNil(), "Invalid argument. input.MachineDeployment can't be nil when calling DeleteAndWaitMachineDeployment")
+
+	log.Logf("Deleting MachineDeployment %s", klog.KObj(input.MachineDeployment))
+	Expect(input.ClusterProxy.GetClient().Delete(ctx, input.MachineDeployment, &client.DeleteOptions{PropagationPolicy: input.DeletePropagationPolicy})).To(Succeed())
+
+	log.Logf("Waiting for MD to be deleted")
+	Eventually(func(g Gomega) {
```

Comment:
q: should we also check that all the machines are gone? (so we check one layer more of the hierarchy)

Reply:
Done